frontend/app/mocks/node.ts

```diff
@@ -1,21 +1,68 @@
+import jsonpatch, { type Operation } from 'fast-json-patch';
 import { HttpResponse, http } from 'msw';
 import { setupServer } from 'msw/node';
+import { z } from 'zod';
+
+import { db } from '~/mocks/db';
+
+/**
+ * Retrieves a user entity based on the provided user ID.
+ *
+ * @param id - The user ID to look up in the database.
+ * @returns The user entity if found, otherwise throws a 404 error.
+ */
+function getUserEntity(id: string | readonly string[]) {
+  const parsedUserId = z.string().uuid().safeParse(id);
+  const userEntity = !parsedUserId.success
+    ? undefined
+    : db.user.findFirst({
+        where: { id: { equals: parsedUserId.data } },
```

Comment:
💩

Reply:
Resolving.. but I still think the formatting is bad.

Reply:
Fix it yourself later.